### PR TITLE
Data Validation: Deep checks

### PR DESCRIPTION
**Pull request by JennyonOort**
Hi Team, 

Below is a summary list of deep-checks added for data validation:

- class imbalance
- null values
- outlier samples
- data duplicates
- mixed data types
- feature label correlation
- feature feature correlation

Please kindly review and let me know if any issues!

### CODE REVIEW

**Comment by javiermtzo99** [comment]
Hi Jenny, I was wondering if you could attach a screenshot of the different data validation cells along with their outputs. That would be really helpful!

**Review by javiermtzo99 — COMMENTED** [review]
I think it would be great to add a text cell with a checklist of our data validations in the analysis.ipynb, similar to what we did in this issue: https://github.com/UBC-MDS/diabetes_predictor_py/issues/18

**Comment by JennyonOort** [comment]
> I think it would be great to add a text cell with a checklist of our data validations in the analysis.ipynb, similar to what we did in this issue: #18

Thanks for the feedback. See reply in Github Issue #18.

**Comment by JennyonOort** [comment]
> Hi Jenny, I was wondering if you could attach a screenshot of the different data validation cells along with their outputs. That would be really helpful!

Sure thing. Another way can search by keyword "validate" as I have added a comment for each validation I have done so should all have this keyword.

**Comment by JennyonOort** [comment]
> I think it would be great to add a text cell with a checklist of our data validations in the analysis.ipynb, similar to what we did in this issue: #18

Checked with Tiffany on this one. We should NOT include the checklist in our analysis.

File: analysis/diabetes_analysis.ipynb
Comment (InderKhera) [inline analysis/diabetes_analysis.ipynb:1]: All Deep Check validation tests work. Deep check validation properly blended in with the pairwise correlation tests conducted from last week.

**Review by InderKhera — APPROVED** [review]
All Deep Check validation tests work. Deep check validation properly blended in with the pairwise correlation tests conducted from last week.

**Pull request merged**